### PR TITLE
Quickstart update for transforms

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**

1) The compiling circuit page in the quick start is separated in two parts:
- PennyLane circuit compilation: `compiling_circuits.rst`
- Workflow compilation (Catalyst): `compiling_workflows.rst`

2) We also add a new quickstart page containing the definition of quantum transformation, examples, composability and relevant document links. `transforming_circuits.rst`

### CODE REVIEW

File: doc/introduction/transforming_circuits.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+* A transform accepts a :class:`~.QuantumTape` as its primary input and
+  returns a sequence of :class:`~.QuantumTape` and an associated processing function.
+
+To streamline the creation of transforms and ensure their versatility across various circuit abstractions in PennyLane
+(:class:`~.QuantumTape`, :class:`~.QNode` and quantum functions), we have created a simple
+decorator :func:`pennylane.transform` that can be applied on your quantum transform respecting the above contract.
+
+Creating your own transform
+---------------------------
```
Comment (josh146) [inline doc/introduction/transforming_circuits.rst:24]:
```suggestion
Creating your own transform
---------------------------

To streamline the creation of transforms and ensure their versatility across
various circuit abstractions in PennyLane, the
:func:`pennylane.transform` is available.

This decorator registers transforms that accept a :class:`~.QuantumTape`
as its primary input, and returns a sequence of :class:`~.QuantumTape`
and an associated processing function.
```

File: doc/introduction/transforming_circuits.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    from typing import Sequence, Callable
+    from pennylane.tape import QuantumTape
+
+    def remove_rx(tape: QuantumTape) -> (Sequence[QuantumTape], Callable):
+
+        operations = filter(lambda op: op.name != "RX", tape.operations)
+        new_tape = type(tape)(operations, tape.measurements, shots=tape.shots)
+
+        def null_postprocessing(results):
+            return results[0]
```
Comment (josh146) [inline doc/introduction/transforming_circuits.rst:42]: Why do you have to index? For me, this isn't a null function, I would expect `lambda *args: args` to be a null function 🤔

File: doc/introduction/transforming_circuits.rst [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+        return [new_tape], null_postprocessing
```
Comment (josh146) [inline doc/introduction/transforming_circuits.rst:44]: A feature request: I would love if I could just do `return [new_tape], None` and have it automatically use a null processing function.

I would also love if I could do:

```python
return new_tape, None
or?
return new_tape
```

for a single circuit->circuit transform.

File: doc/introduction/transforming_circuits.rst [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+            qml.RY(y, wires=0)
+            qml.RZ(y, wires=0)
+            qml.RY(x, wires=0)
+            return qml.expval(qml.PauliZ(wires=0))
+
+In this example, inverses are canceled, leading to the removal of two Hadamard gates. Subsequently, rotations are
+merged into a single :class:`qml.Rot` gate. Consequently, two transforms are successfully applied to the circuit.
+
+Relevant links
+--------------
```
Comment (josh146) [inline doc/introduction/transforming_circuits.rst:95]:
```suggestion
Additional information
----------------------
```

File: doc/introduction/transforming_circuits.rst [hunk 5]
```diff
@@ -0,0 +1,100 @@
+.. role:: html(raw)
+   :format: html
+
+.. _intro_ref_transform_circuits:
+
+Transforming circuits
+=====================
```
[the remark this answers is not in the document]
Comment (trbromley) [reply doc/introduction/transforming_circuits.rst:7]: Yeah good question :thinking: One argument against merging this into "compiling circuits" could be that the result feels a bit long.

Could this content just be at the bottom of [`qml.transforms`](https://xanaduai-pennylane--4897.com.readthedocs.build/en/4897/code/qml_transforms.html)?

[the remark this answers is not in the document]
Comment (josh146) [reply doc/introduction/transforming_circuits.rst:7]: Oh yeah, that's a great idea!

[the remark this answers is not in the document]
Comment (rmoyard) [reply doc/introduction/transforming_circuits.rst:7]: I agree with you I will move it to the transforms page, thanks!

File: doc/introduction/transforming_circuits.rst (hunk 3 above)
Comment (rmoyard) [reply doc/introduction/transforming_circuits.rst:52]: I think that would be a nice improvement for users indeed. Could we add that to the plan for Q1 @trbromley ?

**Review by josh146 — APPROVED** [review]
🙌

**Review by trbromley — APPROVED** [review]
Thanks @rmoyard!

**Pull request merged**